spec/requests/api/v1/keywords_controller_spec.rb

```diff
@@ -58,89 +58,179 @@
-        post api_v1_keywords_path, params: params, headers: create_token_header
+        it 'returns the upload_success meta message' do
+          params = { 'file' => fixture_file_upload('csv/valid.csv'), 'source' => 'google' }
+          post api_v1_keywords_path, params: params, headers: create_token_header
 
-        expect(JSON.parse(response.body)['meta']).to eq(I18n.t('csv.upload_success'))
-      end
+          expect(JSON.parse(response.body)['meta']).to eq(I18n.t('csv.upload_success'))
+        end
+
+        context 'when Google search is successfully fetched' do
+          it 'saves keyword "Apple" with top ads count as 3' do
+            stub_request(:get, %r{google.com/search}).to_return(body: file_fixture('html/valid_google.html').read)
+            params = { 'file' => fixture_file_upload('csv/valid.csv'), 'source' => 'google' }
+            perform_enqueued_jobs do
+              post api_v1_keywords_path, params: params, headers: create_token_header
+            end
 
-      context 'when Google search is successfully fetched' do
-        it 'saves keyword "Apple" with top ads count as 3' do
-          stub_request(:get, %r{google.com/search}).to_return(body: file_fixture('html/valid_google.html').read)
-          params = { 'file' => fixture_file_upload('csv/valid.csv') }
-          perform_enqueued_jobs do
-            post api_v1_keywords_path, params: params, headers: create_token_header
+            expect(Keyword.where(name: 'Apple').first[:ads_top_count]).to eq(3)
           end
+        end
+
+        context 'when Google search returns too many attemps status 422' do
+          it 'does not saves information for "Apple"' do
+            stub_request(:get, %r{google.com/search}).to_return(status: 422)
+            params = { 'file' => fixture_file_upload('csv/valid.csv'), 'source' => 'google' }
+            perform_enqueued_jobs do
+              post api_v1_keywords_path, params: params, headers: create_token_header
+            end
 
-          expect(Keyword.where(name: 'Apple').first[:ads_top_count]).to eq(3)
+            expect(Keyword.where(name: 'Apple').first[:ads_top_count]).to eq(0)
+          end
         end
       end
 
-      context 'when Google search returns too many attemps status 422' do
-        it 'does not saves information for "Apple"' do
-          stub_request(:get, %r{google.com/search}).to_return(status: 422)
-          params = { 'file' => fixture_file_upload('csv/valid.csv') }
-          perform_enqueued_jobs do
-            post api_v1_keywords_path, params: params, headers: create_token_header
-          end
+      context 'when CSV file is missing' do
+        it 'returns unprocessable_entity status' do
+          post api_v1_keywords_path, headers: create_token_header
 
-          expect(Keyword.where(name: 'Apple').first[:ads_top_count]).to eq(0)
+          expect(response).to have_http_status(:unprocessable_entity)
+        end
+
+        it 'returns a invalid_file error' do
```

Comment:
```suggestion
        it 'returns invalid_file error' do
```

Reply:
This should be `returns an invalid_file error`, right?

Reply:
solved in c69e184